scipy/interpolate/_bspl.pyx

```diff
@@ -223,6 +223,42 @@ def evaluate_all_bspl(const double[::1] t, int k, double xval, int m, int nu=0):
     return bbb[:k+1]
 
 
+def insert(double xval,
+           const double[::1] t,
+           const double_or_complex[:, ::1] c,
+           int k,
+        ):
+    """Insert a single knot at `xval`.
+    """
+    cdef:
+        int interval, i
+
+    interval = find_interval(t, k, xval, k, False)
+    if interval < 0:
+        raise ValueError("cannot happen.")
```

Comment:
Since out-of-bounds values are guarded for at the call site at `BSpline.insert_knot`, the only way `interval < 0` is that `xval` is nan. Rephrased in the last commit.